delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/Resolvers.scala

```diff
@@ -0,0 +1,292 @@
+package ch.epfl.bluebrain.nexus.delta.sdk
+
+import akka.persistence.query.{NoOffset, Offset}
+import cats.effect.Clock
+import ch.epfl.bluebrain.nexus.delta.rdf.IriOrBNode.Iri
+import ch.epfl.bluebrain.nexus.delta.sdk.model.identities.Caller
+import ch.epfl.bluebrain.nexus.delta.sdk.model.identities.Identity.Subject
+import ch.epfl.bluebrain.nexus.delta.sdk.model.projects.ProjectRef
+import ch.epfl.bluebrain.nexus.delta.sdk.model.resolvers.IdentityResolution.{ProvidedIdentities, UseCurrentCaller}
+import ch.epfl.bluebrain.nexus.delta.sdk.model.resolvers.ResolverCommand._
+import ch.epfl.bluebrain.nexus.delta.sdk.model.resolvers.ResolverEvent._
+import ch.epfl.bluebrain.nexus.delta.sdk.model.resolvers.ResolverRejection._
+import ch.epfl.bluebrain.nexus.delta.sdk.model.resolvers.ResolverState.{Current, Initial}
+import ch.epfl.bluebrain.nexus.delta.sdk.model.resolvers.ResolverValue.CrossProjectValue
+import ch.epfl.bluebrain.nexus.delta.sdk.model.resolvers._
+import ch.epfl.bluebrain.nexus.delta.sdk.model.{Envelope, Label}
+import fs2.Stream
+import monix.bio.{IO, Task, UIO}
+
+/**
+  * Operations for handling resolvers
+  */
+trait Resolvers {
+
+  /**
+    * Create a new resolver where the id is either present on the payload or self generated
+    * @param project        the project where the resolver will belong
+    * @param resolverFields the payload to create the resolver
+    */
+  def create(project: ProjectRef, resolverFields: ResolverFields)(implicit
+      caller: Caller
+  ): IO[ResolverRejection, ResolverResource]
+
+  /**
+    * Create a new resolver with the provided id
+    * @param id             the resolver identifier
+    * @param project        the project where the resolver will belong
+    * @param resolverFields the payload to create the resolver
+    */
+  def create(id: Iri, project: ProjectRef, resolverFields: ResolverFields)(implicit
```

Comment:
As you can see in `Resources` this is not going to be `id: Iri` but `id: IdSegment`